src/pysource/kernels.py

```diff
@@ -60,6 +65,83 @@ def acoustic_kernel(model, u, fw=True, q=None):
     return pde
 
 
+def SLS_2nd_order(model, u1, u2, fw=True, q=None, qborn=None, f0=0.015):
+    """
+    Viscoacoustic 2nd SLS wave equation
+
+    Parameters
+    ----------
+    model: Model
+        Physical model
+    u1 : TimeFunction
+        Pressure field
+    u2 : TimeFunction
+        Attenuation Memory variable
+    fw: Bool
+        Whether forward or backward in time propagation
+    q : TimeFunction or Expr
+        Full time-space source as a tuple (one value for each component)
+    qborn: Virtual source
+    f0 : Peak frequency
+    """
+    qp, vp = model.qp, model.vp
+
+    damp = model.damp
+
+    q = q or 0
+
+    qborn = qborn or 0
+
+    b = model.irho
+
+    s = model.grid.stepping_dim.spacing
+
+    # The stress relaxation parameter
+    t_s = (sqrt(1.+1./qp**2)-1./qp)/f0
+
+    # The strain relaxation parameter
+    t_ep = 1./(f0**2*t_s)
+
+    # The relaxation time
+    tt = (t_ep/t_s)-1.
+
+    # Density
+    rho = 1. / b
+
+    # Bulk modulus
+    bm = rho * vp**2
+
+    p = u1
+    r = u2
+
+    if fw:
+
+        pde_r = r + s * (tt / t_s) * rho * div(b * grad(p, shift=.5), shift=-.5) - \
+            s * (1. / t_s) * r
+
+        u_r = Eq(r.forward, damp * pde_r)
+
+        pde_p = 2. * p - damp * p.backward + s**2 * bm * (1. + tt) * \
+            div(b * grad(p, shift=.5), shift=-.5) - s**2 * vp**2 * \
+            r.forward + s**2 * vp**2 * qborn
+
+        u_p = Eq(p.forward, damp * pde_p)
```

Comment:
We decided to keep the time discretized implementation as in Devito because back then the `solve` doesn't work to the first-order adjoint equation. 
So, as we intend to include in the future the first-order equations in JUDI too, we decided to use the time discretized implementation to keep the standard.

Reply:
The `solve` works for first order in devito and if it doesn't it should be fixed. The standard should be to use `solve`.

Reply:
See https://github.com/devitocodes/devito/pull/1879

Reply:
Okay. I will check it and make the changes.

Reply:
@mloubout from your PR #1879  in devito, we replaced the equation using hand derivation by `solve`.  Indeed the equation becomes better for user understanding. It's closer to math notation.
Furthermore, we also did all the other suggested corrections.